odl/trafos/convolution.py

```diff
@@ -0,0 +1,849 @@
+                    ''.format(domain, ft_op.domain))
+            self.__ft_op = ft_op
+
+        # Calculate scaled kernel FT
+        kernel_kwargs = kwargs.pop('kernel_kwargs', {})
+        kernel_is_scaled = kwargs.pop('kernel_is_scaled', False)
+        scaled_ker_ft, scaling, ker_ft_op = _scaled_kernel_ft(
+            kernel, self.ft_op, kernel_mode, kernel_kwargs, kernel_is_scaled)
+
+        self.__scaled_kernel_ft = scaled_ker_ft
+        self.__kernel_scaling = scaling
+        self.__kernel_ft_op = ker_ft_op
+
+        # FT range temporary
+        ft_range_tmp = kwargs.pop('ft_range_tmp', None)
+        if ft_range_tmp is not None and ft_range_tmp not in self.ft_op.range:
+            raise OpRangeError('`ft_range_tmp` not in `ft_op.range`')
+        self.__ft_range_tmp = ft_range_tmp
+
+        # Cache adjoint operator
+        self.__adjoint = FourierSpaceConvolution(
+            domain=self.range, range=self.domain,
+            kernel=self.__scaled_kernel_ft.conj(), kernel_mode='fourier',
+            kernel_is_scaled=True, axes=axes, ft_impl=self.ft_impl,
+            integral=self.integral, ft_kwargs=self.__ft_kwargs,
+            ft_range_tmp=self.__ft_range_tmp)
+
+    @property
+    def integral(self):
+        """``True`` if the conv. integral is approximated, else ``False``"""
+        return self.__integral
+
+    @property
+    def ft_op(self):
+        """Fourier transform operator used in this convolution."""
+        return self.__ft_op
+
+    @property
+    def ft_impl(self):
+        """Implementation of the Fourier transform in this convolution."""
+        return self.ft_op.impl
+
+    @property
+    def axes(self):
+        """Axes along which the convolution is taken."""
+        return self.ft_op.axes
+
+    @property
+    def kernel_ft_op(self):
+        """The Fourier transform operator for the kernel.
+
+        Its domain may differ from that of `ft_op`, but the ranges of both
+        transforms are the same.
+        """
+        # TODO: Update docstring when broadcasting convolution is there
+        return self.__kernel_ft_op
+
+    def kernel_ft(self, out=None):
+        """Return a the FT of this convolution operator's kernel.
+
+        Parameters
+        ----------
+        out : element of ``kernel_ft_op.range``, optional
+            Write the output to this data storage.
+
+        Returns
+        -------
+        kernel_ft : element of ``kernel_ft_op.range``
+            FT of the convolution kernel. If ``out`` was given, this
+            object is a reference to it.
+        """
+        if out is None:
+            out = self.kernel_ft_op.range.element()
+
+        out.lincomb(1 / self.__kernel_scaling, self.__scaled_kernel_ft)
+        return out
+
+    def kernel(self, out=None):
+        """Return the real-space kernel of this convolution operator.
+
+        Parameters
+        ----------
+        out : element of ``kernel_ft_op.domain``, optional
```

Comment:
No, the kernel space is usually different from `domain` -- the kernel is most of the time zero-centered, while `domain` can be anything.